### PR TITLE
MM-21356: Conditionally set user status online

### PR DESCRIPTION
**Pull request by agnivade**
#### Summary

Check for the set_online query param and do not set the status
if it is set to false.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-21356

### CODE REVIEW

**Comment by agnivade** [comment]
Good idea, will do.

File: api4/post.go [hunk 1]
```diff
@@ -67,7 +67,15 @@ func createPost(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	c.App.SetStatusOnline(c.App.Session.UserId, false)
+	setOnline := r.URL.Query().Get("set_online")
+	setOnlineBool, err2 := strconv.ParseBool(setOnline)
```
Comment (hanzei) [inline api4/post.go]: I would differ between an empty string and an non empty string here. Then we could log an error if an non empty, but malformed string is received.

File: api4/post.go [hunk 2]
```diff
@@ -67,7 +68,20 @@ func createPost(c *Context, w http.ResponseWriter, r *http.Request) {
...
 	}
 
-	c.App.SetStatusOnline(c.App.Session.UserId, false)
+	setOnline := r.URL.Query().Get("set_online")
+	setOnlineBool := true // By default, always set online.
+	var err2 error
+	if setOnline != "" {
+		setOnlineBool, err2 = strconv.ParseBool(setOnline)
+		if err2 != nil {
+			mlog.Error(err2.Error())
```
Comment (hanzei) [inline api4/post.go]: I don't think we should log an error here. A warning is enough, as the system still works fine. There should also be an explanation, why the errors is shown in the logs.
```suggestion
			mlog.Warn("Failed to parse set_online url query parameter from createPost request", mlog.Err(err2))
```

**Pull request merged**